### PR TITLE
Dependency update action

### PR DESCRIPTION
**Pull request by rettigl**
Adds a github action to update the poetry.lock file weekly using a crone timed action. 
Sets up a pull request for review, if packages are updated, and lists the poetry update output in the PR description.
See previous test workflow runs and closed PRs for examples.

### CODE REVIEW

File: .github/workflows/update_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        uses: packetcoders/action-setup-cache-python-poetry@main
+        with:
+          python-version: 3.8
+          poetry-version: 1.2.2
+
+      # update poetry lockfile
+      - name: "Update poetry lock file"
+        id: update
+        run: |
+          poetry run poetry self update
```
Comment (zain-sohail) [inline .github/workflows/update_dependencies.yml:34]: I dont think a poetry run is needed. poetry self update should work fine.

File: .github/workflows/update_dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          exec 5>&1
+          UPDATE_OUTPUT=$(poetry run poetry update|tee >(cat - >&5))
```
Comment (zain-sohail) [inline .github/workflows/update_dependencies.yml:36]: also here just poetry update would suffice

File: .github/workflows/update_dependencies.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          echo "UPDATE_OUTPUT<<EOF" >> $GITHUB_OUTPUT
+          echo "$UPDATE_OUTPUT" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+          STATUS=$(git status|tee >(cat - >&5))
+          echo "STATUS<<EOF" >> $GITHUB_OUTPUT
```
Comment (zain-sohail) [inline .github/workflows/update_dependencies.yml:42]: I under that these statements are just capturing the packages that are updated.

File: .github/workflows/update_dependencies.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          echo "$STATUS" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
```
Comment (zain-sohail) [inline .github/workflows/update_dependencies.yml:45]: This command will allow us to have the requirements.txt updated

`poetry export --without-hashes --format=requirements.txt -o docs/requirements.txt -E docs -E notebook`

I believe you could put this as another step. and then we delete the generate-requirements.yml

**Review by zain-sohail — CHANGES_REQUESTED** [review]
Is there a github actions run + PR that showcases the complete updating and merging behavior? 
I only found runs which tried updating but had nothing to update, and PRs which you closed without merging.

**Comment by rettigl** [comment]
> Is there a github actions run + PR that showcases the complete updating and merging behavior? I only found runs which tried updating but had nothing to update, and PRs which you closed without merging.

https://github.com/OpenCOMPES/sed/actions/runs/6646883557/job/18061211259 was a successful run that produced this PR:
https://github.com/OpenCOMPES/sed/pull/216
I then updated the lockfile manually, and ran another run, 
https://github.com/OpenCOMPES/sed/actions/runs/6647035812/job/18061860224, which still changed some package names in the lock file. This created this PR: https://github.com/OpenCOMPES/sed/pull/218 which I merged.
Afterwards, I initiated this run: https://github.com/OpenCOMPES/sed/actions/runs/6647161677/job/18062064014 which now correctly does not issue a PR, because the lockfile was not updated.

File: .github/workflows/update_dependencies.yml (hunk 1 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml:34]: We need to do this in the context of the poetry environment, no? Maybe its not needed, but it works

File: .github/workflows/update_dependencies.yml (hunk 2 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml:36]: again, we need to act on the poetry environment. At least here I think we need the poetry run. Anyways, it works...

File: .github/workflows/update_dependencies.yml (hunk 3 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml:42]: They are collecting the updated files in the repository, depending on which a PR is opened or not.

File: .github/workflows/update_dependencies.yml (hunk 4 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml:45]: yes, this should work, I guess.

File: .github/workflows/update_dependencies.yml (hunk 1 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml]: You are right, it's not needed

File: .github/workflows/update_dependencies.yml (hunk 2 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml]: same here

File: .github/workflows/update_dependencies.yml (hunk 4 above)
Comment (rettigl) [reply .github/workflows/update_dependencies.yml:52]: done

**Review by zain-sohail — APPROVED** [review]
Looking quite good. I also like that it can be activated based on multiple criteria.

**Pull request merged**